packages/browser-wallet/src/background/window-management.ts

```diff
@@ -160,3 +166,16 @@ export const forwardToPopup = <P, R>(
 
     bgMessageHandler.handleMessage(createMessageTypeFilter(messageType), conditionalHandler);
 };
+
+export function runConditionComposer<R>(
+    firstHandler: RunCondition<R>,
+    secondHandler: RunCondition<R>
+): RunCondition<R> {
+    return async (msg, sender) => {
+        const first = await firstHandler(msg, sender);
+        if (!first.run) {
+            return first;
+        }
+        return secondHandler(msg, sender);
+    };
```

Comment:
Couldn't we improve this to take an arbitrary number of `RunCondition`s?

Reply:
We could, but why?

Reply:
Well... more reusable with no overhead? What's not to like? Anyway if you don't want to, idc 😉

Reply:
I dont want to do this when we dont need it.